### PR TITLE
feat(designer): add test infrastructure

### PR DESCRIPTION
**Pull request by rachmlenig**
### **PR Type**
Tests, Enhancement

___

### **Description**
- Comprehensive test infrastructure with Vitest, React Testing Library, and MSW

- 775-line projectService API test suite covering CRUD operations and edge cases

- Mock data factories for projects, datasets, and embedding strategies

- MSW handlers for API mocking and test setup configuration

- Detailed testing documentation and npm test scripts

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Test Setup<br/>vitest.config.ts<br/>setup.ts"] --> B["Mock Infrastructure<br/>MSW handlers<br/>Mock server"]
  B --> C["Mock Factories<br/>projectFactory<br/>datasetFactory<br/>embeddingStrategyFactory"]
  C --> D["Test Suites<br/>projectService.test.ts<br/>App.test.tsx"]
  A --> E["Test Utils<br/>renderWithProviders<br/>createTestQueryClient"]
  E --> D
  F["Documentation<br/>TESTING.md<br/>README.md"] -.-> D
  G["package.json<br/>test scripts"] -.-> A
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><details><summary>9 files</summary><table>
<tr>
  <td><strong>projectService.test.ts</strong><dd><code>Comprehensive projectService API test suite</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/507/files#diff-e3c1cf8ca5e9df776875b5aada0286293a95e4c77d93dbe18b2e1eb1106e5213">+775/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>projectFactory.ts</strong><dd><code>Mock project data factory functions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/507/files#diff-a5a666783e5f82ff4b6383c44af91584de33ea6fa4d1b02068af536cf76ed8e9">+123/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>datasetFactory.ts</strong><dd><code>Mock dataset data factory functions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/507/files#diff-ee354f03b77406f5c15095eb64ebb636dd497fc25dd05b4acb560e1a786a3d86">+111/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>embeddingStrategyFactory.ts</strong><dd><code>Mock embedding strategy factory functions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/507/files#diff-83fa7e6463dcc78dc9ee1bca77051571b03e74ebff6c789839431714a52eabde">+110/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>handlers.ts</strong><dd><code>MSW request handlers for API mocking</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/507/files#diff-abf5231d15e9616a12ea2dbdad3266e2bdfb6ad1dc8cc05cff59d1031ad42082">+216/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>server.ts</strong><dd><code>MSW server setup for Node environment</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/507/files#diff-9c7c686622b49ae33440c273734662af25d2557a1850895df926b6481d868cd2">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>setup.ts</strong><dd><code>Global test setup and browser API mocks</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/507/files#diff-b8fb84e2f5e0dadcea3543f4344e16d947b518d792e99f301d528dc2580f0108">+61/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>utils.tsx</strong><dd><code>Custom render utilities and test helpers</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/507/files#diff-4d5bb2772e2b5861abe4a14f199c56edbe65544cca6cbaf95c14e3a2dc44e364">+86/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>App.test.tsx</strong><dd><code>Smoke test for App component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/507/files#diff-4031f3f2deebaaf2783474308664cbf73d9ea149ce5a971ab406846d83e01355">+24/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>vitest.config.ts</strong><dd><code>Vitest configuration with coverage settings</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/507/files#diff-37e2126ea53b698aa27eb744a352644d782ad58d90f993ed6c9e788fbb96d1c1">+32/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>TESTING.md</strong><dd><code>Comprehensive testing guide and best practices</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/507/files#diff-16e45fcab259e8653a932ca4fccd08f998fcee35c31ebdaad8e4477ba0313cc5">+592/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>README.md</strong><dd><code>Test infrastructure directory documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/507/files#diff-12f77ba0677cdf94116c869ca1d640b5e055765e803af85341d3c725a68d6bb5">+108/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>package.json</strong><dd><code>Add test dependencies and npm scripts</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/llama-farm/llamafarm/pull/507/files#diff-24a3bf952e8ace70037412955431f169d560b3c29d41aa85a7351a9416fec173">+14/-2</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

**Comment by rachmlenig** [comment]
I'm going to refactor this a bit. I'll create issues for the test stubs and only commit the completed tests.

File: designer/src/test/setup.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    removeEventListener: vi.fn(),
+    dispatchEvent: vi.fn(),
+  })),
+})
+
+// Start MSW server before all tests
+beforeAll(() => {
+  server.listen({
+    onUnhandledRequest: 'warn',
+  })
```
[the remark this answers is not in the document]
Comment (rachmlenig) [reply designer/src/test/setup.ts:47]: using 'warn' while completing test coverage
